homeassistant/components/onvif/event.py

```diff
@@ -167,14 +167,25 @@ async def async_parse_messages(self, messages) -> None:
                     UNHANDLED_TOPICS.add(topic)
                 continue
 
-            event = await parser(self.unique_id, msg)
+            event: Event = await parser(self.unique_id, msg)
 
             if not event:
                 LOGGER.warning("Unable to parse event from %s: %s", self.unique_id, msg)
                 return
 
+            if self._events.get(event.uid) is None:
+                LOGGER.info(
+                    "Registered %s as a(n) %s with unique id: %s",
+                    event.name,
+                    event.platform,
+                    event.uid,
+                )
+
             self._events[event.uid] = event
 
+            if event.platform == "event" and event.enabled:
+                self.hass.bus.fire(CONF_ONVIF_EVENT, {CONF_UNIQUE_ID: event.uid})
```

Comment:
We're in async context here so we should use the async api.
```suggestion
                self.hass.bus.async_fire(CONF_ONVIF_EVENT, {CONF_UNIQUE_ID: event.uid})
```